crates/nu-command/src/filters/take/take_.rs

```diff
@@ -138,7 +146,7 @@ fn first_helper(
                 }
             }
             _ => {
-                if rows_desired == 1 {
+                if rows_desired == 1 && rows.is_none() {
```

Comment:
@rgwood @fdncred Right, fair enough, I agree merging makes sense. I don't actually agree with this line, since I don't think we should ever do the `rows_desired == 1` behaviour, but this PR makes things better not worse than the current state.

Reply:
Ah, I just copy the code from `first` command, I may need to take a look in detail :-)